File: tasking/sources/task_priority.1.c
/*
* @@name:       task_priority.1c
* @@type:       C
* @@compilable: yes
* @@linkable:   no
* @@expect:     success
* @@version:    omp_4.5
*/
void compute_array (float *node, int M);

void compute_matrix (float *array, int N, int M)
{
   int i;
   #pragma omp parallel private(i)
   #pragma omp single
   {
      for (i=0;i<N; i++) {
         #pragma omp task priority(i)
         compute_array(&array[i*M], M);
      }
   }
}

File: tasking/sources/task_priority.1.f90
! @@name:	task_priority.1
! @@type:	F-free
! @@compilable:	yes
! @@linkable:	no
! @@expect:	success
! @@version:    omp_4.5
subroutine compute_matrix(matrix, M, N)
   implicit none
   integer :: M, N
   real :: matrix(M, N)
   integer :: i
   interface
      subroutine compute_array(node, M)
      implicit none
      integer :: M
      real :: node(M)
      end subroutine
   end interface
   !$omp parallel private(i)
   !$omp single
   do i=1,N
      !$omp task priority(i)
      call compute_array(matrix(:, i), M)
      !$omp end task
   enddo
   !$omp end single
   !$omp end parallel
end subroutine compute_matrix
